desc/equilibrium/utils.py

```diff
@@ -181,3 +183,100 @@ def parse_axis(axis, NFP=1, sym=True, surface=None):
     else:
         raise TypeError("Got unknown axis type {}".format(axis))
     return axis
+
+
+def scale_profile(profile, inner_rho):
+    """Return a new profile whose rho=1 val is value of the old profile at inner_rho.
+
+    Args
+    ----
+        profile (Profile): Profile to scale
+        inner_rho (float): rho to take as the new LCFS rho for the new profile
+    Returns
+    -------
+        new_profile (Profile): new profile whose last value is the inner_rho value of
+        the original profile
+    """
+    params_list = []
+    modes_list = []
+    if isinstance(profile, PowerSeriesProfile):
+        for coeff, mode in zip(profile.params, profile.basis.modes):
+            l = mode[0]
+            params_list.append(coeff * inner_rho**l)
+            modes_list.append(l)
+        new_profile = PowerSeriesProfile(params=params_list, modes=modes_list)
+    elif isinstance(profile, SplineProfile):
+        n_knots = profile.params.size
+        new_knots = np.linspace(0, 1, n_knots, endpoint=True)
+        new_vals = profile(np.linspace(0, inner_rho, n_knots, endpoint=True))
+        new_profile = SplineProfile(
+            values=new_vals, knots=new_knots, method=profile._method
+        )
+    else:
+        raise NotImplementedError(
+            f"Expected PowerSeriesProfile or SplineProfile, got {type(profile)}"
+        )
+    return new_profile
+
+
+def contract_equilibrium(eq, inner_rho):
+    """Create a new equilibrium by using an inner surface of the passed-in equilibrium.
+
+    Args
+    ----
+        eq (Equilibrium): Equilibrium to contract.
+        inner_rho (float): rho value (<1) to contract the Equilibrium to
+
+    Returns
+    -------
+        eq_inner: New Equilibrium object, contracted from the old one such that
+            eq.pressure(rho=inner_rho) = eq_inner.pressure(rho=1), and
+            eq_inner LCFS = eq's rho=inner_rho surface.
+            Note that this will not be in force balance, and so must be re-solved.
+    """
+    errorif(not (inner_rho < 1 and inner_rho > 0), ValueError, "inner_rho should be <1")
```

Comment:
```suggestion
    errorif(not (inner_rho < 1 and inner_rho > 0), ValueError, f"inner_rho should be between  0 and 1 (not included), but got {inner_rho}")
```
the error message doesn't match what are you checking